src/main/java/io/github/oliviercailloux/y2018/apartments/valuefunction/ApartmentValueFunction.java

```diff
@@ -74,27 +45,79 @@
    * to set those two.
    */
   public ApartmentValueFunction() {
-    this.floorAreaValueFunction = new ConstantValueFunction<>(0d);
-    this.nbBedroomsValueFunction = new ConstantValueFunction<>(0d);
-    this.nbSleepingValueFunction = new ConstantValueFunction<>(0d);
-    this.nbBathroomsValueFunction = new ConstantValueFunction<>(0d);
-    this.terraceValueFunction = new ConstantValueFunction<>(0d);
-    this.floorAreaTerraceValueFunction = new ConstantValueFunction<>(0d);
-    this.wifiValueFunction = new ConstantValueFunction<>(0d);
-    this.pricePerNightValueFunction = new ConstantValueFunction<>(0d);
-    this.nbMinNightValueFunction = new ConstantValueFunction<>(0d);
-    this.teleValueFunction = new ConstantValueFunction<>(0d);
-
-    this.floorAreaSubjectiveValueWeight = 0.1;
-    this.nbBedroomsSubjectiveValueWeight = 0.1;
-    this.nbSleepingSubjectiveValueWeight = 0.1;
-    this.nbBathroomsSubjectiveValueWeight = 0.1;
-    this.terraceSubjectiveValueWeight = 0.1;
-    this.floorAreaTerraceSubjectiveValueWeight = 0.1;
-    this.wifiSubjectiveValueWeight = 0.1;
-    this.pricePerNightSubjectiveValueWeight = 0.1;
-    this.nbMinNightSubjectiveValueWeight = 0.1;
-    this.teleSubjectiveValueWeight = 0.1;
+    this.booleanValueFunctions = new EnumMap<>(Criterion.class);
+    this.doubleValueFunctions = new EnumMap<>(Criterion.class);
+    Arrays.stream(Criterion.values())
+        .forEach(
+            criterion -> {
+              if (criterion.hasBooleanDomain()) {
+                setInternalBooleanValueFunction(criterion, new ConstantValueFunction<>(0.0d));
+              } else {
+                setInternalDoubleValueFunction(criterion, new ConstantValueFunction<>(0.0d));
+              }
+            });
+    this.weight = new EnumMap<>(Criterion.class);
+    Arrays.stream(Criterion.values())
+        .forEach(criterion -> setWeightSubjectiveValue(criterion, 0.1d));
+  }
+
+  /**
+   * Used to retrieve a partial boolean value function according to the <code>criterion</code>
+   * passed as a parameter
+   *
+   * @param criterion the criterion associated with the <code>PartialValueFunction</code> that we
+   *     want to obtain
+   * @return the partial boolean value function associated with the <code>criterion</code> parameter
+   */
+  private PartialValueFunction<Boolean> getInternalBooleanValueFunction(Criterion criterion) {
+    checkNotNull(criterion);
+    checkArgument(criterion.hasBooleanDomain());
+    return this.booleanValueFunctions.get(criterion);
```

Comment:
Tant qu’on y est, un petit `verify` que le get ne renvoie pas null serait pleinement rassurant.

Reply:
Fix in 756de746ab31d7fe0ba7a00d3a19418d2adab900